java-logging-appinsights/src/main/java/uk/gov/hmcts/reform/logging/appinsights/telemetry/initializers/CloudRoleNameInitializer.java

```diff
@@ -0,0 +1,17 @@
+package uk.gov.hmcts.reform.logging.appinsights.telemetry.initializers;
+
+import com.microsoft.applicationinsights.extensibility.context.ContextTagKeys;
+import com.microsoft.applicationinsights.telemetry.Telemetry;
+import com.microsoft.applicationinsights.web.extensibility.initializers.WebTelemetryInitializerBase;
+
+public class CloudRoleNameInitializer extends WebTelemetryInitializerBase {
+    @Override
+    protected void onInitializeTelemetry(Telemetry telemetry) {
+        if (SpringApplicationName.get() == null) {
+            throw new IllegalStateException("spring.application.name configuration property is not set");
+        } else {
+            telemetry.getContext().getTags().put(ContextTagKeys.getKeys().getDeviceRoleName(),
```

Comment:
There's a more understandable approach in terms of reading this setting... setting:

```java
telemetry.getContext().getDevice().setRoleName(SpringApplicationName.get());
```